core/src/main/java/com/alamkanak/weekview/ViewState.kt

```diff
@@ -31,19 +31,21 @@ internal class ViewState {
 
     private var isFirstDraw: Boolean = true
 
-    // Drawing context
-    private var startPixel: Float = 0f
-    val startPixels: MutableList<Float> = mutableListOf()
-    val dateRange: MutableList<Calendar> = mutableListOf()
-    val dateRangeWithStartPixels: MutableList<Pair<Calendar, Float>> = mutableListOf()
-
     // Calendar configuration
     var numberOfVisibleDays: Int = 3
     var restoreNumberOfVisibleDays: Boolean = true
     var showFirstDayOfWeekFirst: Boolean = false
     var showCurrentTimeFirst: Boolean = false
     var arrangeAllDayEventsVertically: Boolean = true
 
+    // Drawing context
+    private var startPixel: Float = 0f
+    val startPixels: MutableList<Float> = mutableListOf()
+    val dateRange: MutableList<Calendar> = mutableListOf<Calendar>().apply {
+        addAll(createDateRange(firstVisibleDate).validate(this@ViewState))
+    }
```

Comment:
This could be slightly more concise.

```suggestion
    val dateRange: MutableList<Calendar> = createDateRange(firstVisibleDate).validate(this).toMutableList()
```

Reply:
Thanks, amended